### DISCUSSION TITLE
Uncaught FFI\Exception: Failed loading 'libvips.so.42'

### DISCUSSION THREAD

**Discussion by mirudo2** [libvips/libvips#4172] [Q&A] [answered]
```php
$ffi = \FFI::cdef("
    double sin(double x);
", "libm.so.6"); // 200 OK

$ffi = \FFI::cdef(<<<'CPP'
            int vips_init (const char *argv0);
            const char *vips_error_buffer (void);
            int vips_version(int flag);
        CPP, "libvips.so.42"); // error 500
```
      
I don't know what the hell I'm doing wrong, but it seems that FFI can't load the specific library "libvips.so.42", both are in the same directory "/usr/lib/x86_64-linux-gnu/", "libm.so.6" is loaded normally, but "libvips.so.42" is not.

I'm on Ubuntu 22.04, php 8.0.30, libvips 8.11.4 (I've tried several versions)

I've been trying to fix this for only 2 days.

**Comment by jcupitt** [comment]
Hello @mirudo2,

The `libvips.so.42` file is a symlink made by the `libvips-dev` package. Without it, you'll just have the so with the exact ABI number, eg. `libvips.so.42.18.0`.

You might also need to run `ldconfig` I guess?

I imagine you've tried php-vips, but it has some extra library finding code, it might be worth testing as well.

I'll make a dockerfile for current php-vips and libvips when I'm back at my desk tomorrow.

**Reply by mirudo2** [reply]
Hello, Thanks for replying,
Yes, I have already run `sudo ldconfig`,
I also noticed that `libvips.so.42` is just a symbolic link,
And I have also tried to load the file directly, with the absolute path:

```php
$ffi = \FFI::cdef(<<<'CPP'
int vips_init (const char *argv0);
const char *vips_error_buffer (void);
int vips_version(int flag);
CPP, "/usr/local/lib/libvips.so.42.13.3");
```
At first I thought it could be a problem with the FFI extension, so I did a test:

```php
$ffi = \FFI::cdef("
double sin(double x);
", "libm.so.6");

echo $ffi->sin(1.0); // 0.8414709848079
```
It seems that the extension is not the problem.
Maybe some setting in php.ini?

```php
extension=ffi.so

ffi.enable = true
ffi.allow_c_call = true
ffi.allow_unsafe_globals = true
```

This is php(8.0.30) from Xampp, in the last case I will try to install a newer version

**Reply by mirudo2** [reply, marked as answer]
Hi, @jcupitt,
The problem has been solved!
It seems that the Xampp version of php was the problem,
After installing php and nginx on the system, without Xampp,
there is no longer a 500 error when running the script above.
Thanks

**Reply by jcupitt** [reply]
Ah great. Thanks for the update.
